src/pack.c

```diff
@@ -939,7 +939,7 @@ git_off_t get_delta_base(
 			if (left <= used)
 				return GIT_EBUFS;
 			base_offset += 1;
-			if (!base_offset || MSB(base_offset, 7))
```

Comment:
To me, it looks like the author of this code already assumed `base_offset` being unsigned. Otherwise `base_offset += 1; if (!base_offset) /* Overflow */;` wouldn't make any sense at all. So I think using `size_t` instead and then having a check whether its value is greater than the maximum value for `git_off_t` should be good.

Reply:
Using `size_t` now.